### PR TITLE
Ml exercise first revision

### PR DESCRIPTION
**Pull request by nanistyarp**

### CODE REVIEW

File: nanis/ml_exercises/scripts/week_3.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    intial_gradient = (1/number_of_samples) * x.T @ (predictions-y) 
+    regularisation = (lambda_value/number_of_samples) * np.r_[[[0]], theta[1:].reshape(-1,1)]
+    gradient = initial_gradient + regularisation
+    return gradient.flatten()
+
+
+if __name__ == '__main__':
+    # PART 1
+    file_path = 'data/ex2data1.txt'
+    variables = ['score_1', 'score_2', 'admitted']
```
Comment (anthony-khong) [inline nanis/ml_exercises/scripts/week_3.py:104]: This should really be called column_names

File: nanis/ml_exercises/scripts/week_3.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    exam_data = clean_data(file_path, variables)
+
+    xlabel = ('exam_score_1')
+    ylabel = ('exam_score_2')
+    plot_data(exam_data, xlabel, ylabel)
+
+    x, y = extract_features(exam_data)
+    number_of_samples, features = x.shape
```
Comment (anthony-khong) [inline nanis/ml_exercises/scripts/week_3.py:112]: number_of_features

File: nanis/ml_exercises/scripts/week_3.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    theta = np.zeros(features)
+
+    cost = compute_cost(theta, x, y)
+    optimised_cost_function = optimised_cost(theta, x, y)
+
+    threshold = 0.5
+    exam_score_samples = np.array([1, 45, 85])
+    optimised_theta = optimised_cost_function.x
```
Comment (anthony-khong) [inline nanis/ml_exercises/scripts/week_3.py:120]: This should be wrapped in another abstraction layer. What is `x`?

File: nanis/ml_exercises/scripts/week_3.py [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    sigmoid(np.array(exam_score_samples).dot(optimised_theta.T))
```
Comment (anthony-khong) [inline nanis/ml_exercises/scripts/week_3.py:121]: What is this for?

File: nanis/ml_exercises/scripts/week_3.py [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+    # PART 2 
+    file_path = 'data/ex2data2.txt' 
+    variables = ['test_2', 'test_2', 'pass']
+    microchip_data = clean_data(file_path, variables)
+
+    xlabel = ('microchip test 1')
+    ylabel = ('microchip test 2')
+    plot_data(microchip_data, xlabel, ylabel)
+
+    x, y = extract_features(microchip_data)
```
Comment (anthony-khong) [inline nanis/ml_exercises/scripts/week_3.py:136]: Avoid single-letter variable names especially when the scope is more than one line.

File: nanis/ml_exercises/scripts/week_3.py (hunk 4 above)
Comment (nanistyarp) [reply nanis/ml_exercises/scripts/week_3.py:121]: This is a section to test if the function is working properly. Should have assigned to sample_admission_probability

File: nanis/ml_exercises/scripts/week_3.py (hunk 3 above)
Comment (nanistyarp) [reply nanis/ml_exercises/scripts/week_3.py:120]: x is the solution for theta from scipy.optimize.minimize

**Pull request closed without merging**